src/main/scala/uk/gov/nationalarchives/omega/api/business/records/GetRecordService.scala

```diff
@@ -0,0 +1,328 @@
+    val relatedToMap = getLabelledIdentifierMap(relatedToEntities)
+    val separatedFromMap = getLabelledIdentifierMap(separatedFromEntities)
+    val subjectMap = getSubjectMap(uriSubjectEntities, labelledSubjectEntities)
+    recordSummaryEntities.map(summary =>
+      RecordDescriptionFull(
+        summary = RecordDescriptionSummary(
+          identifier = GeneralIdentifier.fromUri(summary.recordDescriptionUri),
+          secondaryIdentifier = secondaryIdentifierMap.get(summary.recordDescriptionUri),
+          label = summary.scopeAndContent,
+          description = summary.scopeAndContent,
+          accessRights = accessRightsMap.getOrElse(summary.recordDescriptionUri, List.empty),
+          isPartOf = isPartOfMap.getOrElse(summary.recordDescriptionUri, List.empty),
+          previousSibling = summary.previousSiblingUri.flatMap(uri => Some(GeneralIdentifier(uri))),
+          versionTimestamp = summary.versionTimestamp,
+          previousDescription = summary.previousDescriptionUri.flatMap(uri => Some(GeneralIdentifier(uri)))
+        ),
+        properties = getRecordPropertiesEntity(
+          recordPropertiesEntities,
+          summary.recordDescriptionUri,
+          isReferencedByMap,
+          relatedToMap,
+          separatedFromMap,
+          subjectMap
+        )
+      )
+    )
+  }
+
+  private def getAccessRightsMap(accessRightsEntities: List[AccessRightsEntity]): Map[String, List[Identifier]] = {
+    val groupedMap: Map[String, List[AccessRightsEntity]] =
+      accessRightsEntities.groupBy(_.recordDescriptionUri)
+    groupedMap.map(entry => entry._1 -> entry._2.map(a => GeneralIdentifier(a.accessRights)))
+  }
+
+  private def getIsPartOfMap(isPartOfEntities: List[IsPartOfEntity]): Map[String, List[Identifier]] =
+    isPartOfEntities
+      .groupBy(_.recordDescriptionUri.toString)
+      .map(entry => entry._1 -> entry._2.map(a => GeneralIdentifier.fromUri(a.recordSetConceptUri)))
+
+  private def getSecondaryIdentifiersMap(
+    secondaryIdentifierEntities: List[SecondaryIdentifierEntity]
+  ): Map[String, List[TypedIdentifier]] =
+    secondaryIdentifierEntities
+      .groupBy(_.recordDescriptionUri.toString)
+      .map(entry =>
+        entry._1 -> entry._2.map(a => TypedIdentifier(GeneralIdentifier(a.secondaryIdentifier), a.identifierProperty))
+      )
+
+  private def getLabelledIdentifierMap(
+    labelledIdentifierEntities: List[LabelledIdentifierEntity]
+  ): Map[String, List[LabelledIdentifier]] =
+    labelledIdentifierEntities
+      .groupBy(_.recordDescriptionUri.toString)
+      .map(entry => entry._1 -> entry._2.map(a => LabelledIdentifier(a.identifier.toString, a.label)))
+
+  private def getSubjectMap(
+    uriSubjectEntities: List[IdentifierEntity],
+    labelledSubjectEntities: List[LabelledIdentifierEntity]
+  ): Map[String, List[Identifier]] = {
+    import cats.implicits._
+    val map1: Map[String, List[Identifier]] = uriSubjectEntities
```

Comment:
I am not sure about the naming `map1`, `map2`, `map3`. It seems a bit abstract to me.

Reply:
Renamed.